lib/parslet/atoms/sequence.rb

```diff
@@ -29,9 +29,10 @@ def try(source, context, consume_all)
       success, value = p.apply(source, context, child_consume_all) 
 
       unless success
-        return context.err(self, source, @error_msgs[:failed], [value]) 
+        return context.err(self, source, @error_msgs[:failed], [value])
       end
-      
+      context.succ(source)
```

Comment:
I find it odd that only the sequence calls 'context.succ' - should we not call context.succ from 'Base#succ' for all successes? What am I missing here?